store/stage.go

```diff
@@ -158,41 +160,59 @@ func (s *Store) findStageRaw(ctx context.Context, find *api.StageFind) ([]*stage
 }
 
 // createStageImpl creates a new stage.
-func (*Store) createStageImpl(ctx context.Context, tx *Tx, create *api.StageCreate) (*stageRaw, error) {
-	query := `
+func (*Store) createStageImpl(ctx context.Context, tx *Tx, creates []*api.StageCreate) ([]*stageRaw, error) {
+	var valueStr []string
+	var values []interface{}
+	for i, create := range creates {
+		values = append(values,
+			create.CreatorID,
+			create.CreatorID,
+			create.PipelineID,
+			create.EnvironmentID,
+			create.Name,
+		)
+		const count = 5
+		valueStr = append(valueStr, fmt.Sprintf("($%d,$%d,$%d,$%d,$%d)", i*count+1, i*count+2, i*count+3, i*count+4, i*count+5))
+	}
+
+	query := fmt.Sprintf(`
 		INSERT INTO stage (
 			creator_id,
 			updater_id,
 			pipeline_id,
 			environment_id,
 			name
-		)
-		VALUES ($1, $2, $3, $4, $5)
-		RETURNING id, creator_id, created_ts, updater_id, updated_ts, pipeline_id, environment_id, name` + `
-	`
-	var stageRaw stageRaw
-	if err := tx.QueryRowContext(ctx, query,
-		create.CreatorID,
-		create.CreatorID,
-		create.PipelineID,
-		create.EnvironmentID,
-		create.Name,
-	).Scan(
-		&stageRaw.ID,
-		&stageRaw.CreatorID,
-		&stageRaw.CreatedTs,
-		&stageRaw.UpdaterID,
-		&stageRaw.UpdatedTs,
-		&stageRaw.PipelineID,
-		&stageRaw.EnvironmentID,
-		&stageRaw.Name,
-	); err != nil {
-		if err == sql.ErrNoRows {
-			return nil, common.FormatDBErrorEmptyRowWithQuery(query)
+		) VALUES %s
+		RETURNING id, creator_id, created_ts, updater_id, updated_ts, pipeline_id, environment_id, name ORDER BY id ASC`, strings.Join(valueStr, ","))
```

Comment:
The `ORDER BY` here fails the test. Looks like the `INSERT..RETURNING..` statement doesn't support the `ORDER BY`, and we also don't need it either. https://dba.stackexchange.com/questions/95822/does-postgres-preserve-insertion-order-of-records